src/sentry/issues/related/__init__.py

```diff
@@ -13,9 +13,10 @@
 }
 
 
-def find_related_issues(group: Group) -> list[dict[str, list[int] | str]]:
-    related_issues: list[dict[str, list[int] | str]] = []
+def find_related_issues(group: Group) -> list[dict[str, str | list[int] | dict[str, str]]]:
```

Comment:
For here it definitly looks like we can use a typed dict:
```python
class RelatedIssueItem(TypedDict):
  meta: something
  type: str
  data: something
  
def find_related_issues(group: Group) -> list[RelatedIssueItem]
```
But I'll leave the specific types to you!